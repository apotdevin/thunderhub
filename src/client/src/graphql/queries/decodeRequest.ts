import { gql } from '@apollo/client';

export const DECODE_REQUEST = gql`
  query DecodeRequest($request: String!) {
    decodeRequest(request: $request) {
      chain_address
      cltv_delta
      description
      description_hash
      destination
      destination_node {
        node {
          alias
        }
      }
      expires_at
      id
      routes {
        base_fee_mtokens
        channel
        cltv_delta
        fee_rate
        public_key
      }
      tokens
    }
  }
`;
