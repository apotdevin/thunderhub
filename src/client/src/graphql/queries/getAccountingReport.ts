import { gql } from '@apollo/client';

export const GET_ACCOUNTING_REPORT = gql`
  query GetAccountingReport(
    $category: String
    $currency: String
    $fiat: String
    $month: String
    $year: String
  ) {
    getAccountingReport(
      category: $category
      currency: $currency
      fiat: $fiat
      month: $month
      year: $year
    )
  }
`;
