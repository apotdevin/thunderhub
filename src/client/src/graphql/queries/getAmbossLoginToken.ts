import { gql } from '@apollo/client';

export const GET_AMBOSS_LOGIN_TOKEN = gql`
  query GetAmbossLoginToken($redirect_url: String) {
    getAmbossLoginToken(redirect_url: $redirect_url)
  }
`;
