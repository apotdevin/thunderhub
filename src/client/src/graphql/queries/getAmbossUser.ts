import { gql } from '@apollo/client';

export const GET_AMBOSS_USER = gql`
  query GetAmbossUser {
    getAmbossUser {
      subscription {
        end_date
        subscribed
        upgradable
      }
      backups {
        last_update
        last_update_size
        total_size_saved
        available_size
        remaining_size
      }
      ghost {
        username
      }
    }
  }
`;
