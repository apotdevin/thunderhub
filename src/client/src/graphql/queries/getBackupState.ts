import { gql } from '@apollo/client';

export const GET_BACKUP_STATE = gql`
  query GetBackupState {
    getBackupState
  }
`;
