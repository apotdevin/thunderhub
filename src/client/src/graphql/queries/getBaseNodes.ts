import { gql } from '@apollo/client';

export const GET_BASE_NODES = gql`
  query GetBaseNodes {
    getBaseNodes {
      _id
      name
      public_key
      socket
    }
  }
`;
