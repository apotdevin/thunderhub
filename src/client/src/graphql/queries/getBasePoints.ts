import { gql } from '@apollo/client';

export const GET_BASE_POINTS = gql`
  query GetBasePoints {
    getBasePoints {
      alias
      amount
    }
  }
`;
