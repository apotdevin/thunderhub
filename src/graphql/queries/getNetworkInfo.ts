import { gql } from '@apollo/client';

export const GET_NETWORK_INFO = gql`
  query GetNetworkInfo {
    getNetworkInfo {
      averageChannelSize
      channelCount
      maxChannelSize
      medianChannelSize
      minChannelSize
      nodeCount
      notRecentlyUpdatedPolicyCount
      totalCapacity
    }
  }
`;
