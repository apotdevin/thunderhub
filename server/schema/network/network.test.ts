import testServer from 'server/tests/testServer';
import { GET_NETWORK_INFO } from 'src/graphql/queries/getNetworkInfo';

jest.mock('ln-service');

describe('Network Resolvers', () => {
  describe('getNetworkInfo', () => {
    test('success', async () => {
      const { query } = testServer();

      const res = await query({
        query: GET_NETWORK_INFO,
        variables: { auth: { type: 'test' } },
      });

      expect(res.errors).toBe(undefined);
      expect(res).toMatchSnapshot();
    });
  });
});
