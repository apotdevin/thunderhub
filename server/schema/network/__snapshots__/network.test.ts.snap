// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Network Resolvers getNetworkInfo success 1`] = `
Object {
  "data": Object {
    "getNetworkInfo": Object {
      "averageChannelSize": "1000",
      "channelCount": 123,
      "maxChannelSize": 456,
      "medianChannelSize": 789,
      "minChannelSize": 100,
      "nodeCount": 10000,
      "notRecentlyUpdatedPolicyCount": 10,
      "totalCapacity": "10000",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
